packages/core/test/logger.js

```diff
@@ -210,52 +235,88 @@ describe('logger', () => {
         "something": ":censored:6:57a71b6062:",
         "somethingElse": ":censored:6:e3b0ee5182:",
       }`,
-          request_url: 'https://test.com/?api_key=:censored:8:89250e9365:',
+          request_url: 'https://test.com/?api_key=:censored:11:94aca9077b:',
         },
       },
     ]);
   });
 
-  it('should replace sensitive data inside response', async () => {
+  it('should replace novel sensitive data', async () => {
+    // this test should, as closely as possible, match what we actually log after an http request from z.request
     const bundle = {
       authData: {
-        refresh_token: 'whatever',
+        refresh_token: 'very_secret',
       },
     };
     const logger = createlogger({ bundle }, options);
 
-    const data = {
-      response_json: {
```

Comment:
the crux of why this test wasn't catching what it was supposed to was that when we log an http request, the body for the `req` and `res` is a string. So `response_json` being here let us find the secrets when we actually don't. If you removed this property, the test failed (hence the logged secrets)